tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -7314,21 +7392,8 @@ bool HLSLExternalSource::CanConvert(
           goto lSuccess;
         }
         break;
-      }
-    }
-
-    if (const BuiltinType *BT = source->getAs<BuiltinType>()) {
-      BuiltinType::Kind kind = BT->getKind();
-      switch (kind) {
-      case BuiltinType::Kind::UInt:
-      case BuiltinType::Kind::Int:
-      case BuiltinType::Kind::Float:
-      case BuiltinType::Kind::LitFloat:
-      case BuiltinType::Kind::LitInt:
-        if (explicitConversion) {
-          Second = ICK_Flat_Conversion;
-          goto lSuccess;
-        }
```

Comment:
This conditional and switch is a duplicate of the previous. It depends on the same variables for the result and changes none of them. So the repetition accomplishes nothing.

Reply:
😲

Reply:
Good catch!